prov/gni/src/gnix_cntr.c

```diff
@@ -219,14 +243,15 @@ static int gnix_cntr_wait_sleep(struct gnix_fid_cntr *cntr_priv,
 {
 	int ret = FI_SUCCESS;
 
-	while (atomic_get(&cntr_priv->cnt) < threshold) {
-		ret = gnix_wait_wait((struct fid_wait *)cntr_priv->wait,
+	while (atomic_get(&cntr_priv->cnt) < threshold &&
+	       atomic_get(&cntr_priv->cnt_err) == 0) {
+		ret = fi_wait((struct fid_wait *)cntr_priv->wait,
```

Comment:
I was just following the example set in the psm provider, but I can switch it to a direct call to gnix_wait_wait instead.